engine/src/api/services/llm_client.py

```
@@ -23,27 +29,186 @@ class LLMResponse:
+                )
+
+        # Configure host and API key for self-hosted/alternative endpoints via LiteLLM
+        if self.host:
+            try:
+                if self.api_key:
+                     litellm.set_api_key(self.api_key, self.llm_provider) # Set key if available
+                litellm.api_base = self.host
+                logger.info(f"Configured LiteLLM API base for {self.llm_provider} to: {self.host}")
+            except Exception as e:
+                logger.error(f"Failed to configure self-hosted model API base: {str(e)}")
+                raise RuntimeError(f"Failed to configure self-hosted model API base: {str(e)}") from e
```

Ruff (0.11.9)

Line too long (102 > 100)

(E501)